Modules/Core/Common/src/itkMersenneTwisterRandomVariateGenerator.cxx

```diff
@@ -175,11 +171,11 @@ MersenneTwisterRandomVariateGenerator::Initialize(const IntegerType seed)
   // See Knuth TAOCP Vol 2, 3rd Ed, p.106 for multiplier.
   // In previous versions, most significant bits (MSBs) of the seed affect
   // only MSBs of the state array.  Modified 9 Jan 2002 by Makoto Matsumoto.
-  IntegerType * s = m_State;
-  IntegerType * r = m_State;
+  IntegerType *       s = m_State;
+  const IntegerType * r = m_State;
 
   *s++ = seed;
-  for (IntegerType i = 1; i < MersenneTwisterRandomVariateGenerator::StateVectorLength; ++i)
+  for (IntegerType i = 1; i < StateVectorLength; ++i)
   {
     *s++ = uint32_t{ (uint32_t{ 1812433253 } * (*r ^ (*r >> 30)) + i) };
     ++r;
```

Comment:
Honestly it's not clear to me why it uses two pointers, `s` and `r`, instead of one. 🤷 Both pointers always just point to the same element of `m_State`.

Reply:
Maybe kick out `r` pointer and leave only `s`?

Reply:
I just hesitated because the original C++ implementation appeared to have both an `s` and an `r`. At least, if this is the original implementation: https://www.cs.rpi.edu/academics/courses/fall12/ds/hw/05_multi_linked_lists/MersenneTwister.h

	register uint32 *s = state;
	register uint32 *r = state;
	register int i = 1;
	*s++ = seed & 0xffffffffUL;
	for( ; i < N; ++i )
	{
		*s++ = ( 1812433253UL * ( *r ^ (*r >> 30) ) + i ) & 0xffffffffUL;
		r++;
	}

But we did already make quite some style improvements compared to that one, anyway.